runtime/include/chplmemtrack.h

```diff
@@ -29,6 +29,8 @@
 #include <stdlib.h>
 
 
+///// These entry points support debugging.
+
```

Comment:
You could characterize the first set as supporting the memory tracking user interface in MemTracking.chpl, and the second set as being the behind-the-scenes data collection machinery. (The latter are called at the top level of the memory allocation interface, which is used by both the compiler emitted code and the runtime.)

I'm not getting the bit about dead code removal -- neither the C compiler nor even the linker can dead-code remove globally visible functions.